src/distributions/uniform.rs

```diff
@@ -769,9 +779,12 @@ impl UniformSampler for UniformChar {
 
     fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> Self::X {
         let mut x = self.sampler.sample(rng);
-        if x >= 0xD800 {
-            x += 0xE000 - 0xD800;
+        if x >= CHAR_SURROGATE_START {
+            x += CHAR_SURROGATE_LEN;
```

Comment:
Shouldn't this use rejection sampling?

Reply:
That would be slower. We know exactly what the number of available values are.

Reply:
Actually, we do use rejection sampling, within the int-range sampler.